containers/cpp-mariadb/test-project/main.cpp

```diff
@@ -0,0 +1,68 @@
+/*-------------------------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See https://go.microsoft.com/fwlink/?linkid=2090316 for license information.
+ *-------------------------------------------------------------------------------------------------------------*/
+
+#include <iostream>
+#include <mariadb/conncpp.hpp>
+
+using namespace std;
+
+int main()
+{
+    cout << "Hello, Remote World!" << endl;
+
+    try
+    {
+        string databasename = getenv("MARIADB_DATABASE");
+        string password = getenv("MARIADB_PASSWORD");
+        string username = getenv("MARIADB_USER");
+        string hostname = getenv("MARIADB_DATABASE");
+        string jointURL = "jdbc:mariadb://" + hostname + "/" + databasename;
+
+        // Instantiate Driver
+        sql::Driver* driver = sql::mariadb::get_driver_instance();
+
+        cout << "DB Connecting" << endl;
+
+        // Configure Connection
+        sql::SQLString url(jointURL);
+        sql::Properties properties({{"user", username}, {"password", password}});
+
+        // Establish Connection
+        unique_ptr<sql::Connection> conn(driver->connect(url, properties));
```

Comment:
conn requires checking for null. See https://mariadb.com/docs/clients/mariadb-connectors/connector-cpp/connect/#sql-driver-connect. 

A null pointer would not result in a `sql::SQLException` and thus crash the program

Reply:
hi @juzuluag, my $0.02 is that here (and above for the `try`), that xtophs is suggesting we either:

**(a)** do what this link suggested and test the `conn` for null: https://mariadb.com/docs/clients/mariadb-connectors/connector-cpp/connect/#sql-driver-connect

**(b)** or, we do what this page suggested and  catch the `sql::SQLException`: https://mariadb.com/docs/clients/mariadb-connectors/connector-cpp/connect/#code-example-connect